core/templates/components/graph-services/graph-layout.service.spec.ts

```diff
@@ -0,0 +1,1251 @@
+      'State1',
+      'State2',
+      'State3',
+      'State4',
+      'State5'
+    ];
+
+    expect(sgls.getIndentationLevels(adjacencyLists, longestPathIds)).toEqual(
+      [0, 0.5, 1, 0, 0]);
+
+    let shortestPathIds: string[] = [
+      'State1',
+      'State4',
+      'State5'
+    ];
+    expect(sgls.getIndentationLevels(adjacencyLists, shortestPathIds)).toEqual(
+      [0, 0, 0]);
+  });
+
+  it('should not return indentation level greater' +
+    ' than MAX_INDENTATION_LEVEL', () => {
+    let adjacencyLists = {
+      State1: ['State2', 'State3'],
+      State2: ['State3', 'State6'],
+      State6: ['State3', 'State8'],
+      State8: ['State3', 'State7'],
+      State7: ['State3', 'State9'],
+      State9: ['State3', 'State10'],
+      State10: ['State3'],
+      State3: ['State5'],
+      State5: []
+    };
+
+    let trunkNodeIds: string[] = [
+      'State1',
+      'State2',
+      'State6',
+      'State8',
+      'State7',
+      'State9',
+      'State10',
+      'State3',
+      'State5'
+    ];
+
+    let returnedIndentationLevels = sgls.getIndentationLevels(
+      adjacencyLists, trunkNodeIds);
+    returnedIndentationLevels.forEach(indentationLevel => {
+      expect(indentationLevel).toBeLessThanOrEqual(sgls.MAX_INDENTATION_LEVEL);
+    });
+  });
+
+  it('should return augmented links with bezier curves', () => {
+    let nodeData = {
+      State1: {
+        depth: 0,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.15333333333333335,
+        xLabel: 0.1625,
+        yLabel: 0.23333333333333334,
+        id: 'State1',
+        label: 'State1',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State2: {
+        depth: 1,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.42000000000000004,
+        xLabel: 0.1625,
+        yLabel: 0.5,
+        id: 'State2',
+        label: 'State2',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State3: {
+        depth: 1,
+        offset: 1,
+        reachable: true,
+        x0: 0.29750000000000004,
+        y0: 0.42000000000000004,
+        xLabel: 0.3875,
+        yLabel: 0.5,
+        id: 'State3',
+        label: 'State3',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      },
+      State4: {
+        depth: 2,
+        offset: 0,
+        reachable: true,
+        x0: 0.07250000000000001,
+        y0: 0.6866666666666666,
+        xLabel: 0.1625,
+        yLabel: 0.7666666666666666,
+        id: 'State4',
+        label: 'State4',
+        height: 0.16,
+        width: 0.18000000000000002,
+        reachableFromEnd: false
+      }
+    };
+
+    // Here, bezier curve follows the format 'M%f %f Q %f %f %f %f'. The
+    // floating point values are caluclated with the help of the position values
```

Comment:
"calculated" is misspelled.

As an aside, I think I've seen several misspellings in recent PRs. Please try and find a way to proofread your comments / variable names before submitting for review.

Reply:
Done